bin/cylc-get-host-metrics

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""cylc get-host-metrics [OPTIONS]
+
+   Get metrics for localhost, in the form of a JSON structure with
+   top-level keys as requested via the OPTIONS:
+
+   1. --load
+      1, 5 and 15 minute load averages (as keys) from the 'uptime' command.
+   2. --memory
+      Total free RAM memory, in kilobytes, from the 'free -k' command.
+   3. --disk-space=PATH / --disk-space=PATH1,PATH2,PATH3 (etc)
+      Available disk space from the 'df -Pk' command, in kilobytes, for one
+      or more valid PATHs within the filesystem of localhost. Multiple PATH
+      options can be specified via a comma-delimited list, each becoming a
+      key under the top-level disk space key.
+
+   If no options are specified, --load and --memory are invoked by default."""
+
+
+import sys
+if '--use-ssh' in sys.argv[1:]:
+    sys.argv.remove('--use-ssh')
+    from cylc.remote import remrun
+    if remrun():
+        sys.exit(0)
+
+import os
+import re
+from subprocess import Popen, PIPE, CalledProcessError
+import json
+
+import cylc.flags
+from cylc.option_parsers import CylcOptionParser as COP
+
+
+LOAD_TEMPLATES = ['uptime', (1, 5, 15), r'load average:\s(.+)\n',
+                  r',\s', r'(\d+\.?\d+)', r'\d+\.?\d+']
+MEMORY_TEMPLATES = ['free -k', r'Mem:\s+(.+)\n', r'\d+\s+\d+\s+(\d+)\s+']
+DISK_TEMPLATES = ['df -Pk', r'Mounted on\n\S+\s+(.+)\n',
+                  r'\d+\s+\d+\s+(\d+)\s+']
```

Comment:
Very good idea. I was assuming anyone reading the code would be far more familiar & experienced with regular expressions than I, so that they read almost trivially, but it certainly will not hurt to add some comments.